tests/runtest.sh

```diff
@@ -397,20 +404,23 @@ function precompile_overlay_assemblies {
                     echo Unable to generate dasm for $filename
                 fi
             else
-                # Precompile any assembly except mscorlib since we already have its NI image available.
-                if [[ "$filename" != *"mscorlib.dll"* ]]; then
-                    if [[ "$filename" != *"mscorlib.ni.dll"* ]]; then
-                        echo Precompiling $filename
-                        $overlayDir/crossgen /Platform_Assemblies_Paths $overlayDir $filename 2>/dev/null
-                        local exitCode=$?
-                        if [ $exitCode == -2146230517 ]; then
-                            echo $filename is not a managed assembly.
-                        elif [ $exitCode != 0 ]; then
-                            echo Unable to precompile $filename.
-                        else
-                            echo Successfully precompiled $filename
-                        fi
+                if [[ -v skipCrossGenFiles[$(basename $filename)] ]]; then
+                    continue
+                fi
+                echo Precompiling $filename
+                $overlayDir/crossgen /Platform_Assemblies_Paths $overlayDir $filename 1> $filename.stdout 2>$filename.stderr
+                local exitCode=$?
+                if [[ $exitCode != 0 ]]; then
+                    if grep -q -e '(COR_E_ASSEMBLYEXPECTED)' $filename.stderr; then
```

Comment:
Since exit code is not reliable using grep of stdout to find the not a managed assembly error.  Choose to search on the human readable error rather than the hex error.